gio/src/file_attribute_info.rs

```diff
@@ -13,30 +13,22 @@ glib::wrapper! {
     match fn {
         copy => |ptr| {
             let copy = glib::ffi::g_malloc0(mem::size_of::<ffi::GFileAttributeInfo>()) as *mut ffi::GFileAttributeInfo;
-            if !(*ptr).name.is_null() {
-                (*copy).name = glib::ffi::g_strdup((*ptr).name);
-            }
+            (*copy).name = glib::ffi::g_strdup((*ptr).name);
             copy
         },
         free => |ptr| {
-            if !(*ptr).name.is_null() {
-                glib::ffi::g_free((*ptr).name as *mut _);
-            }
+            glib::ffi::g_free((*ptr).name as *mut _);
```

Comment:
Is `ptr` here guaranteed to be nonnull?

Reply:
Yes

Reply:
That's good then, I looked at the docs and wasn't sure

Reply:
It's only called internally in `glib/src/{shared,boxed,boxed_inline,object}.rs` for the corresponding type, and mostly from the `Drop` impl. All the pointers in use there are required to be non-`NULL`.